### PR TITLE
UI: Small styling fix for viewing certificate in kubernetes configuration

### PR DESCRIPTION
**Pull request by malinac02**
Small styling fix viewing certificate in kubernetes configuration

**Problem:**

- There was a problem when viewing a certificate in Kubernetes configuration. When the Kubernetes CA Certificate is too short, the styling of the certificate card messes up
- There was also a problem where the styling messes up (regardless of the length of the CA Certificate) when the browser resizes
- Design also requested more padding between the certificate text and the copy button to the right
- Before:
Before

**Fix:**
- Created a new component for the certificate card (so that it can be used elsewhere in the future)
- Changed to using new HDS::Card (fixed issues with the browser resizing, fixed the issue with padding)
- Added additional CSS to get rid of original styling issue with too short a certificate
- After:
After

### CODE REVIEW

File: ui/lib/kubernetes/addon/components/page/configuration.hbs [hunk 1]
```diff
@@ -9,32 +9,7 @@
...
-                @clipboardText={{@config.kubernetesCaCert}}
-                @buttonType="button"
-                @success={{action (set-flash-message "Certificate copied")}}
-              >
-                <Icon @name="clipboard-copy" aria-label="Copy" />
-              </CopyButton>
-            </div>
-          </div>
-        </div>
+        <CertificateCard @kubernetesCaCert={{@config.kubernetesCaCert}} />
```
Comment (kiannaquach) [inline ui/lib/kubernetes/addon/components/page/configuration.hbs:12]: If we're making this a general component that lives in core, the argument name should be more general. Maybe an argument called `certificateValue`?

File: ui/tests/integration/components/certificate-card-test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  test('it renders without a CA Certificate', async function (assert) {
+    await render(hbs`<CertificateCard />`);
+
+    assert.dom(SELECTORS.label).hasText('PEM Format', 'The label text is correct');
+    assert.dom(SELECTORS.value).hasNoText('The is no value for the CA Certificate');
+  });
+
+  test('it renders with a small CA Certificate', async function (assert) {
+    await render(hbs`<CertificateCard @kubernetesCaCert="test"/>`);
```
Comment (kiannaquach) [inline ui/tests/integration/components/certificate-card-test.js:27]: If we update the argument name, we would also need to update the tests to pass the correct names :)

File: ui/lib/kubernetes/addon/components/page/configuration.hbs (hunk 1 above)
Comment (kiannaquach) [inline ui/lib/kubernetes/addon/components/page/configuration.hbs:12]:
```suggestion
        <CertificateCard @certificateVault={{@config.kubernetesCaCert}} />
```

Comment (kiannaquach) [reply ui/lib/kubernetes/addon/components/page/configuration.hbs:12]: If we're making this a general component that lives in core, the argument name should be more general. Maybe an argument called certificateValue?

File: ui/tests/integration/components/certificate-card-test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      gbEwga4wHQYDVR0OBBYEFFXI70krXeQDxZgbaCQoR4jUDncEMH8GA1UdIwR4MHaA
+      FFXI70krXeQDxZgbaCQoR4jUDncEoVukWTBXMQswCQYDVQQGEwJDTjELMAkGA1UE
+      CBMCUE4xCzAJBgNVBAcTAkNOMQswCQYDVQQKEwJPTjELMAkGA1UECxMCVU4xFDAS
+      BgNVBAMTC0hlcm9uZyBZYW5nggEAMAwGA1UdEwQFMAMBAf8wDQYJKoZIhvcNAQEE
+      BQADQQA/ugzBrjjK9jcWnDVfGHlk3icNRq0oV7Ri32z/+HQX67aRfgZu7KWdI+Ju
+      Wm7DCfrPNGVwFWUQOmsPue9rZBgO
+      -----END CERTIFICATE-----
+    `;
+    this.set('certificate', certificate);
+    await render(hbs`<CertificateCard @kubernetesCaCert={{this.certificate}}/>`);
```
Comment (kiannaquach) [inline ui/tests/integration/components/certificate-card-test.js:52]: 👆🏻same here w/ the argument

File: ui/app/styles/components/certificate-card.scss
Comment (Monkeychip) [inline ui/app/styles/components/certificate-card.scss:1]: In general, we try and not do component specific styling if possible. Instead we first look to helper classes. 

For the flex-grow part we could use this helper class instead `is-flex-1`. 
And for the width: Is there any chance we could use a percent or maybe css-grid instead of a hardcoded px? Even something like is `is-halfwidth` where the width is 50%. Maybe poke around the helper-classes, specifically the helper-classes/layout.scss or helper-classes/flexbox-and-grid.scss. We can always make a new helper class, which is preferred over a component specific style. Happy to help zoom with you if nothing is super obvious for styling.

File: ui/lib/kubernetes/addon/components/page/configuration.hbs (hunk 1 above)
Comment (malinac02) [reply ui/lib/kubernetes/addon/components/page/configuration.hbs:12]: Thanks for catching that! Will change that!

File: ui/app/styles/components/certificate-card.scss
Comment (malinac02) [reply ui/app/styles/components/certificate-card.scss:1]: Thanks for letting me know! Just changed to using `is-flex-1`. 

The reason I hardcoded in the pixels was because I wanted to make sure CertificateCard remained the same size even when the browser shrinks and resizes. But I never took that up with design! I'll ask Yankun if it should shrink/expand or not (and therefore if I should use something like `is-halfwidth`). Thank you for bringing this up! 👍

Comment (kiannaquach) [reply ui/app/styles/components/certificate-card.scss:1]: I agree with Angel on this^ Using something like `is-halfwidth` and aiming for responsive styling should be prioritized :)

Comment (malinac02) [reply ui/app/styles/components/certificate-card.scss:1]: Sounds good!

Comment (malinac02) [reply ui/app/styles/components/certificate-card.scss:1]: @kiannaquach Yankun ended up saying that the certificate card should remain a fixed size when the browser resizes. I played around with different ways to do this that is in line with the current styling conventions and the CSS RFC. I ended up just adding an `is_medium_width` class in layout.scss, where there already exists an `is_medium_height` and defining it using `$desktop / 3`, which is similar (but not identical) to the pixels I had put before. (And deleted the component CSS file I created.)

@Monkeychip I know you suggested hard coding in another value to _size_variables.scss but since I would have to add a new class somewhere anyways, I skipped doing that so that I wasn't also modifying that file in addition to layout.scss. But if you want me to hard code a size there let me know! 😄 

Long story short I read through the code and CSS RFC a lot looking for the best way to hard code in a size in a way that the team normally does but I'm not sure if I picked the best solution. Let me know if I should change it! (Sorry for such a long message on such a nit-picky thing, just wanna make sure I don't do anything too different from what the rest of the code looks like 😅)

File: ui/app/styles/helper-classes/layout.scss [hunk 4]
```diff
@@ -52,6 +52,10 @@
   min-width: 0;
 }
 
+.is-medium-width {
```
Comment (Monkeychip) [inline ui/app/styles/helper-classes/layout.scss:55]: awesome! thank you.

**Review by Monkeychip — APPROVED** [review]
Thanks for all the back and forth on the styling. We're just starting to get a pattern down so I know the code doesn't make our "best-practice" obvious.
